examples/python/bpsraw.py

```diff
@@ -0,0 +1,423 @@
+    start = 0
+    for i in range(p[0]):
+        for j in range(p[1]):
+            for k in range(p[2]):
+                if [i, j, k] == list(i_rank):
+                    return start
+                m_nodes = GlobalNodes(p, (i, j, k), degree, mesh_elem)
+                start += m_nodes[0] * m_nodes[1] * m_nodes[2]
+    return -1
+
+def CreateRestriction(ceed, mesh_elem, P, num_comp, ndofs, offsets):
+    return ceed.ElemRestriction(mesh_elem, P, num_comp, 1, ndofs, offsets, cmode=libceed.USE_POINTER)
+
+# def CreateRestriction(ceed, mesh_elem, P, num_comp):
+#     num_elem = mesh_elem[0] * mesh_elem[1] * mesh_elem[2]
+#     m_nodes = [mesh_elem[d] * (P - 1) + 1 for d in range(3)]
+
+#     # Allocate idx array of length num_elem * P^3
+#     idx = np.empty(num_elem * P * P * P, dtype=np.int64)
+#     idx_p = 0
+#     for i in range(mesh_elem[0]):
+#         for j in range(mesh_elem[1]):
+#             for k in range(mesh_elem[2]):
+#                 for ii in range(P):
+#                     for jj in range(P):
+#                         for kk in range(P):
+#                             # k,j,i ordering
+#                             idx[idx_p] = num_comp * (((i*(P-1) + ii) * m_nodes[1] + (j*(P-1) + jj)) * m_nodes[2] + (k*(P-1) + kk))
+#                             idx_p += 1
+#     l_size = m_nodes[0] * m_nodes[1] * m_nodes[2] * num_comp
+#     return ceed.ElemRestriction(num_elem, P * P * P, num_comp, 1, l_size, idx, cmode=libceed.USE_POINTER)
+
+# Read command line options
+def parse_args():
+    parser = argparse.ArgumentParser(description="CEED BPs in PETSc using Python")
+    parser.add_argument('-problem', type=int, default=1, help='Polynomial degree (P)')
+    parser.add_argument('-degree', type=int, default=1, help='Polynomial degree (P)')
+    parser.add_argument('-q_extra', type=int, default=1, help='Extra quadrature points (Q-P)')
+    parser.add_argument('-ceed', type=str, default='/cpu/self', help='libCEED resource')
+    parser.add_argument('-local', type=int, default=1000, help='Local nodes per MPI rank')
+    parser.add_argument('-test', action='store_false', help='Run analytic test')
+    parser.add_argument('-benchmark', action='store_false', help='Print performance stats')
+    args, petsc_argv = parser.parse_known_args()
+    sys.argv = [sys.argv[0]] + petsc_argv
+    return args
+
+# PETSc Shell context class
+class MatMassCtx:
+    def __init__(self, ceed, op_mass, ndofs, comm):
+        self.ceed = ceed
+        self.op = op_mass
+        self.ndofs = ndofs
+        self.comm = comm
+        self._xglob = np.zeros(ndofs, dtype=float)
+        self.u = ceed.Vector(ndofs)
+        self.v = ceed.Vector(ndofs)
+
+    def mult(self, A: PETSc.Mat, x: PETSc.Vec, y: PETSc.Vec):
+        lo, hi = x.getOwnershipRange()
+        self._xglob.fill(0.0)
+        self._xglob[lo:hi] = x.array_r
+        self.u.set_array(self._xglob,cmode=libceed.USE_POINTER)
+        self.v.set_value(0.0)

```

Comment:
```suggestion
        self.v.set_value(0.0)

```
this shouldn't be needed